### ISSUE TITLE
Add optimizer to `ChainedScheduler`

### ISSUE THREAD

**Issue by rohitgr7** [pytorch/pytorch#67601]
## 🐛 Bug

Similar to #67318 we should have an optimizer argument for `ChainedScheduler` for consistency and compatibility.
https://github.com/pytorch/pytorch/blob/97f29bda59deab8c063cf01f0a8ff4321b93c55e/torch/optim/lr_scheduler.py#L760

## To Reproduce

Simialar example as in #67318
```py
import torch
model = torch.nn.Linear(2,2)
optim = torch.optim.SGD(model.parameters(), lr=0.1)
lr_scheduler = torch.optim.lr_scheduler.ChainedScheduler([torch.optim.lr_scheduler.ConstantLR(optim), torch.optim.lr_scheduler.StepLR(optim, step_size=10)])

lr_scheduler.optimizer
```
error:
```py
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'ChainedScheduler' object has no attribute 'optimizer'
```

lr_scheduler.optimizer

## Expected behavior

No error.
Looks like a simple fix :)

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): master
 - OS (e.g., Linux): macos
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.9
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

## Additional context

cc @vincentqb @jbschlosser @albanD @datumbox

**Comment by datumbox** [comment]
Eventually this should properly be fixed to call the constructor of the parent class but an easy fix is to apply the same treatment as in #67406.

**Comment by OliverFM** [comment]
I have opened a PR, #68010, to apply the fix you suggested!
